build_tools/azure/test_pytest_soft_dependency.sh

```diff
@@ -7,6 +7,8 @@ source activate $VIRTUALENV
 conda remove -y py pytest || pip uninstall -y py pytest
 
 if [[ "$COVERAGE" == "true" ]]; then
+    # conda may remove coverage when uninstall pytest and py
+    pip install coverage
```

Comment:
why don't we conda install it? or will that bring pytest back? I'm a bit confused.

Reply:
yes we can conda install it. The soft dependency test only runs when DISTRIB=="conda".

Reply:
pip is generally way faster single package installs in CI..

Reply:
Okay lets go back to pip :D

Reply:
I don't mind either way, it was just a neutral observation )